### PR TITLE
Configurable Client Constants

### LINKED ISSUE #82: Make Default Timeout configurable.
_Originally posted by @RichardKeo in https://github.com/bh2smith/dune-client/pull/72#discussion_r1318598224_

### PR DESCRIPTION
**Pull request by bh2smith**
Making `BASE_URL` and `REQUEST_TIMEOUT` (formerly `DEFAULT_TIMEOUT`) constants configurable through environment. 

Closes #82

## Test Plan

No logical changes - Existing CI should suffice.

cc @RichardKeo

### CODE REVIEW

File: dune_client/base_client.py [hunk 1]
```diff
@@ -16,8 +16,8 @@ class BaseDuneClient:
     and provides some convenient functions to use in other clients
     """
 
-    BASE_URL = "https://api.dune.com"
-    DEFAULT_TIMEOUT = 10
+    BASE_URL = os.environ.get("BASE_API_URL", "https://api.dune.com")
+    REQUEST_TIMEOUT = float(os.environ.get("REQUEST_TIMEOUT", 10))
```
[the remark this answers is not in the document]
Comment (bh2smith) [reply dune_client/base_client.py:20]: Good thinking! Will do.

[the remark this answers is not in the document]
Comment (bh2smith) [reply dune_client/base_client.py]: Done.

File: dune_client/base_client.py [hunk 2]
```diff
@@ -35,7 +39,11 @@ def from_env(cls) -> BaseDuneClient:
         without having to import dotenv or os manually
         We use `DUNE_API_KEY` as the environment variable that holds the API key.
         """
-        return cls(os.environ["DUNE_API_KEY"])
+        return cls(
+            api_key=os.environ["DUNE_API_KEY"],
+            base_url=os.environ.get("BASE_API_URL", "https://api.dune.com"),
+            request_timeout=float(os.environ.get("REQUEST_TIMEOUT", 10)),
```
Comment (RichardKeo) [inline dune_client/base_client.py:45]: Should the environment variables be prefixed with `DUNE_API` to "namespace" them?

**Review by RichardKeo — APPROVED** [review]
Looks good, I've left one last comment about the name of the environment variables to make it slightly more tidy.

File: dune_client/base_client.py (hunk 2 above)
Comment (bh2smith) [reply dune_client/base_client.py:45]: Sure!

**Pull request merged**